python/lsst/verify/compatibility/metricTask.py

```diff
@@ -0,0 +1,196 @@
+
+            - ``measurement``: the value of `getOutputMetric` derived from
+                               ``inputData`` (`lsst.verify.Measurement` or
+                               `None`). The measurement shall contain not only
+                               the value of the metric, but any additional
+                               data needed to place it in the proper context.
+
+        Raises
+        ------
+        lsst.verify.MetricComputationError
+            Raised if an algorithmic or system error prevents calculation
+            of the metric. Examples include corrupted input data or
+            unavoidable exceptions raised by analysis code. The
+            ``MetricComputationError`` should be chained to a more specific
+            exception describing the root cause.
+
+            Note that not having enough data for a metric to be applicable is
+            not an error, and should not trigger this exception.
+
+        Examples
+        --------
+        Consider a metric that characterizes PSF variations across the entire
+        field of view, given processed images. Then, if ``run`` has the
+        signature ``run(images)``:
+
+        .. code-block:: py
+
+            inputData = {'images': [image1, image2, ...]}
+            inputDataIds = {'images': [{'visit': 42, 'ccd': 1},
+                                       {'visit': 42, 'ccd': 2},
+                                       ...]}
+            outputDataId = {'measurement': {'visit': 42}}
+        """
+        result = self.run(**inputData)
+        if result.measurement is not None:
+            self.addStandardMetadata(result.measurement,
+                                     outputDataId["measurement"])
+        return result
+
+    @classmethod
+    @abc.abstractmethod
+    def getInputDatasetTypes(cls, config):
+        """Return input dataset types for this task.
+
+        Parameters
+        ----------
+        config : ``cls.ConfigClass``
+            Configuration for this task.
+
+        Returns
+        -------
+        datasets : `dict` from `str` to `str`
+            Dictionary where the key is the name of the input dataset (must
+            match a parameter to `run`) and the value is the name of its
+            Butler dataset type.
+        """
+
+    @classmethod
+    @abc.abstractmethod
+    def getOutputMetric(cls, config):
+        """Return the metric calculated by this task.
+
+        Parameters
+        ----------
+        config : ``cls.ConfigClass``
+            Configuration for this task. The same ``MetricTask`` class may be
+            configured to calculate different related metrics.
+
+        Returns
+        -------
+        metric : `lsst.verify.Name`
+            The metric computed by objects of this class when configured
+            with ``config``.
+        """
+
+    def addStandardMetadata(self, measurement, outputDataId):
+        """Add dataset-specific metadata required for all metrics.
```

Comment:
That's because this method is, in many ways, a placeholder.

The issue is that measurement clients (including, but not limited to, SQuaSH), may require measurements and/or jobs to have specific metadata that they use in their own processing. What this metadata actually *is* is vaguely specified and subject to change (in fact, the requirements *did* change out from under me while I was working on this ticket). So I need a place to add measurement-level metadata that is required for all measurements (a counterexample would be the way `TimingMetricTask` reports that it uses `pipe.base.timeMethod` as its source of truth), but I have to be vague about exactly what that entails.

If you have a better way to summarize all that, I'm all ears.